central/image/datastore/store/v2/postgres/store.go

```diff
@@ -0,0 +1,1030 @@
+	defer metrics.SetPostgresOperationDurationTime(time.Now(), ops.WalkByQuery, "Image")
+
+	conn, release, err := s.acquireConn(ctx, ops.WalkByQuery, "Image")
+	if err != nil {
+		return err
+	}
+	defer release()
+
+	tx, err := conn.Begin(ctx)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := tx.Rollback(ctx); err != nil {
+			log.Errorf("error rolling back: %v", err)
+		}
+	}()
+
+	ctxWithTx := postgres.ContextWithTx(ctx, tx)
+	fetcher, closer, err := pgSearch.RunCursorQueryForSchema[storage.Image, *storage.Image](ctxWithTx, pkgSchema.ImagesSchema, q, s.db)
+	if err != nil {
+		return err
+	}
+	defer closer()
+	for {
+		rows, err := fetcher(cursorBatchSize)
+		if err != nil {
+			return pgutils.ErrNilIfNoRows(err)
+		}
+		for _, image := range rows {
+			err := s.populateImage(ctx, tx, image)
+			if err != nil {
+				return err
+			}
+			if err := fn(image); err != nil {
+				return err
+			}
+		}
+		if len(rows) != cursorBatchSize {
+			break
+		}
+	}
+	return nil
+}
+
+// GetImageMetadata returns the image without scan/component data.
+func (s *storeImpl) GetImageMetadata(ctx context.Context, id string) (*storage.Image, bool, error) {
+	defer metrics.SetPostgresOperationDurationTime(time.Now(), ops.Get, "ImageMetadata")
+
+	return pgutils.Retry3(ctx, func() (*storage.Image, bool, error) {
+		return s.retryableGetImageMetadata(ctx, id)
+	})
+}
+
+func (s *storeImpl) retryableGetImageMetadata(ctx context.Context, id string) (*storage.Image, bool, error) {
+	conn, release, err := s.acquireConn(ctx, ops.Get, "Image")
+	if err != nil {
+		return nil, false, err
+	}
+	defer release()
+
+	row := conn.QueryRow(ctx, getImageMetaStmt, id)
+	var data []byte
+	if err := row.Scan(&data); err != nil {
+		return nil, false, pgutils.ErrNilIfNoRows(err)
+	}
+
+	var msg storage.Image
+	if err := msg.UnmarshalVTUnsafe(data); err != nil {
+		return nil, false, err
+	}
+	return &msg, true, nil
+}
+
+// GetManyImageMetadata returns images without scan/component data.
+func (s *storeImpl) GetManyImageMetadata(ctx context.Context, ids []string) ([]*storage.Image, []int, error) {
+	defer metrics.SetPostgresOperationDurationTime(time.Now(), ops.GetMany, "Image")
+
+	return pgutils.Retry3(ctx, func() ([]*storage.Image, []int, error) {
+		return s.retryableGetManyImageMetadata(ctx, ids)
+	})
+}
+
+func (s *storeImpl) retryableGetManyImageMetadata(ctx context.Context, ids []string) ([]*storage.Image, []int, error) {
+	if len(ids) == 0 {
+		return nil, nil, nil
+	}
+
+	scopeChecker := sac.GlobalAccessScopeChecker(ctx).AccessMode(storage.Access_READ_ACCESS).Resource(targetResource)
+	scopeTree, err := scopeChecker.EffectiveAccessScope(permissions.ResourceWithAccess{
+		Resource: targetResource,
+		Access:   storage.Access_READ_ACCESS,
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+	sacQueryFilter, err := sac.BuildClusterNamespaceLevelSACQueryFilter(scopeTree)
+	if err != nil {
+		return nil, nil, err
+	}
+	q := search.ConjunctionQuery(
```

Comment:
Why this access scoping isn't done for other getter and walk functions in the store?

Reply:
More logic copied from today.  I will take a look.

Reply:
Frankly I'm not sure why it is checking that anyway.  It is using the Image schema which at the end of the day is namespace and cluster scoped.  I think it is completely redundant as the search framework will build the same.  As you point out if we need it one place, we would need it in all of them.  I'm going to take it out.

Reply:
The PR that introduced it
https://github.com/stackrox/stackrox/pull/2528/files#